### ISSUE TITLE
Using local editor?

### ISSUE THREAD

**Issue by roginfarrer** [mikew/nvrh#15]
After using the utility a bit more, I was a bit confused by what it says it does and what it seems to do. I was expecting (or hoping) that nvrh would use my local Neovim editor (and config and such) on a codebase in a remote server. But it seems like it just sshs into my server, and starts an instance of Neovim there (with its own config files).

My understanding of VSCode's remote development is that it uses your local editor and allows you to navigate the remote codebase, run the language servers there, etc. But it's still your local editor and config. https://github.com/microsoft/vscode-docs/blob/main/docs/remote/ssh.md

I have a maybe somewhat unique situation where my remote codebase is on a super old machine that has very old support for C, so it can't compile Treesitter parsers and other utilities. So I don't really want to use the neovim editor on remote. 

I get if this plugin doesn't provide this functionality, but it might make sense to clarify in the README.

**Comment by mikew** [comment]
Ah, you might be interested in [Distant](https://distant.dev/) and [distant.nvim](https://distant.dev/editors/neovim/). But I think in there, [LSPs also run remotely](https://github.com/chipsenkbeil/distant.nvim/issues/65).

I've never had much luck with it, but you might also want to look into SSHFS. Tho I am curious what the remote machine is doing in your scenario, other than keeping the code there.

---

VSCode Remote is way, way, way better architected. There, plugins get to say whether they should run with the UI (locally) or with the workspace (remotely), and your settings are applied to both local and remote machines (well, your settings are just applied to the relevant plugins, wherever they happen to be running).

If this is the architecture of VSCode Remote

![architecture-ssh](https://github.com/user-attachments/assets/a400bca3-ea30-4600-b0d4-e066d949f6c0)

Here's the architecture of Neovim Remote

![nvrh-architecture-ssh](https://github.com/user-attachments/assets/7789a4c6-f921-48dd-a3c5-a9ab3d2b662b)

Personally, I use remote development to trim down the number of machines I have to keep a config in sync on, and to consolidate my in-progress work and architectures to a single machine. I want the local machine as out-of-the-picture as possible.

**Comment by roginfarrer** [comment]
@mikew I did look into Distant, but I think there are still some rough edges since I ran into a few hitches when trying to set it up. It also somehow created 450GB of cache that took down my machine until I realized it and deleted it all. 

I also think the size of my codebase and age of my remote machine is an issue. (CentOS 7 in a 40GB codebase 😭). Ultimately, it was taking me too much time to figure out how to get Neovim working, so I begrudgingly adopted VSCode so I could actually do work 🥲. We're supposedly updating our VMs to Ubuntu in the next 6 months, so hopefully this becomes a non-issue soon.

Thanks for responding to my inquiries!
